csharp/ql/test/utils/model-generator/theorems/TheoremSummaries.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Summaries;
+
+public class Theorems1<T> {
```

Comment:
We'll also want test cases for `void addMany(IList<T> xs)` and `IList<T> getMany()`. These should result in
```
Arg[0].Element -> Arg[this].SynthField[T]
Arg[this].SynthField[T] -> ReturnValue.Element
```
respectively.

Reply:
I will add these, but I suppose the more realistic versions are `void AddMany(IList<T> xs)` and `IList<T> GetMany()` in `CollectionTheorems<T>`, where they will get summaries like
```
AddMany: Argument[0].Element -> Argument[this].Element
GetMany: Argument[this].Element -> ReturnValue.Element
```

Reply:
Yes. Might as well add both variants as tests.